File: view/android/VerificationCodeJavaDemo/app/src/main/java/com/example/verificationcodejavademo/utils/AESUtil.java
package com.example.verificationcodejavademo.utils;


import android.text.TextUtils;
import android.util.Base64;

import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;

/**
 * Date:2020/5/20
 * author:wuyan
 */
public class AESUtil {
    private final static String ALGORITHMSTR = "AES/ECB/PKCS7Padding";
    private final static String ALGORITHM = "AES";

    public static String encode(String content, String key) {
        if (TextUtils.isEmpty(content)) {
            return content;
        }
        if (key == null || key.isEmpty()) {
            return content;
        }
        try {
            byte[] result = encrypt(key, content);

            String s = Base64.encodeToString(result, Base64.DEFAULT);
            if (s.contains("\n")) {
                return s.replace("\n", "");
            }
            return s;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return "";
    }

    private static byte[] encrypt(String encryptKey, String content) throws Exception {
        // 创建AES秘钥
        SecretKeySpec secretKeySpec = new SecretKeySpec(encryptKey.getBytes(), ALGORITHM);
        // 创建密码器
        Cipher cipher = Cipher.getInstance(ALGORITHMSTR);
        // 初始化加密器
        cipher.init(Cipher.ENCRYPT_MODE, secretKeySpec);
        // 加密
        return cipher.doFinal(content.getBytes("UTF-8"));
    }
}

File: view/android/VerificationCodeDemo/app/src/main/java/com/example/verificationcodedemo/utils/AESUtil.kt
package com.example.verificationcodedemo.utils

import android.provider.Contacts.SettingsColumns.KEY
import android.text.TextUtils
import android.util.Base64
import android.util.Log
import java.security.SecureRandom
import javax.crypto.Cipher
import javax.crypto.KeyGenerator
import javax.crypto.spec.SecretKeySpec

/**
 * Date:2020/5/8
 * author:wuyan
 */
object AESUtil {
    private val ALGORITHMSTR: String = "AES/ECB/PKCS7Padding"
    private val ALGORITHM: String = "AES"

    fun encode(content: String, key: String): String {
        if (TextUtils.isEmpty(content)) {
            return content
        }
        if (key.isEmpty()) {
            return content
        }
        try {
            val result = encrypt(key, content)
            var s = String(Base64.encode(result, Base64.DEFAULT))
            if (s.contains("\n")) {
                return s.replace("\n", "")
            }
            return s
        } catch (e: Exception) {
            e.printStackTrace()
        }

        return ""
    }

    @Throws(Exception::class)
    private fun encrypt(encryptKey: String, content: String): ByteArray {
        // 创建AES秘钥
        val secretKeySpec = SecretKeySpec(encryptKey.toByteArray(), ALGORITHM)
        // 创建密码器
        val cipher = Cipher.getInstance(ALGORITHMSTR)
        // 初始化加密器
        cipher.init(Cipher.ENCRYPT_MODE, secretKeySpec)
        // 加密
        return cipher.doFinal(content.toByteArray(charset("UTF-8")))
    }

}